library/src/blas3/rocblas_trsm.hpp

```diff
@@ -0,0 +1,1743 @@
+                                                                 m,
+                                                                 stride_X,
+                                                                 batch_count);
+                        if(i - BLOCK < 0)
+                            break;
+                        rocblas_gemm_template<BATCHED, !BATCHED>(handle,
+                                                                 transB,
+                                                                 transA,
+                                                                 m,
+                                                                 i,
+                                                                 BLOCK,
+                                                                 &negative_one<T>,
+                                                                 0,
+                                                                 (U)X,
+                                                                 i * m,
+                                                                 m,
+                                                                 stride_X,
+                                                                 A,
+                                                                 i * lda + offset_Ain,
+                                                                 lda,
+                                                                 stride_A,
+                                                                 &one<T>,
+                                                                 0,
+                                                                 B,
+                                                                 offset_Bin,
+                                                                 ldb,
+                                                                 stride_B,
+                                                                 batch_count);
+                    }
+                }
+            }
+        } // tranpsose
+
+        return rocblas_status_success;
+    }
+
+    template <rocblas_int BLOCK, bool BATCHED, typename T, typename U, typename V>
+    rocblas_status special_trsm_template(rocblas_handle    handle,
+                                         rocblas_side      side,
+                                         rocblas_fill      uplo,
+                                         rocblas_operation transA,
+                                         rocblas_diagonal  diag,
+                                         rocblas_int       m,
+                                         rocblas_int       n,
+                                         const T*          alpha,
+                                         rocblas_stride    stride_alpha,
+                                         U                 A,
+                                         rocblas_int       offset_Ain,
+       
```

Comment:
I noticed this shows up twice now, in the _mem and template. Not sure if we need it in both, just thought I'd point it out.

Reply:
I removed the logging in _mem for now since it's only used with the _template right now.
As for the prefix, I'll leave it for now since we log to stderr in other places I believe, but likely a good idea to implement in the future.

Reply:
Logging should only occur in the `_impl` functions.

Reply:
Moved stderr logging out of template function into _impl functions.